### DISCUSSION TITLE
Adblock Plus does pass through geo specific propaganda ads

### DISCUSSION THREAD

**Discussion by andry81** [uBlockOrigin/uAssets#14394] [General]
I don't know where to put it, but I think I have to put it into the light by the extension which works as expected.

The Adblock Plus does not block Ukraine war propaganda in Russia (Mozilla Firefox extension: https://addons.mozilla.org/en-US/firefox/addon/adblock-plus).
There is evidences, for example, this link: https://linuxhint.com/bash_-print-_variable/
It might be not visible if browser thinks you are from a different country.

I've tested the link and actually it does replace one ads by propaganda ads, which means it behaves like a malware.
I don't think this is a random issue and adblockplus did it on purpose, so all addons like these: https://github.com/easylist/ruadlist/wiki/RU-Adlist-links may be also compromised.

**Comment by MasterKia** [comment]
Could you send screenshot from before and after the ads is swapped?
Also uBlock Origin doesn't have the ability to "swap" ads, so you shouldn't have to worry about any compromise in that regard.

**Reply by andry81** [reply]
> Could you send screenshot from before and after the ads is swapped?

<details>
  <summary>https://linuxhint.com/bash_-print-_variable/</summary>
<img src="https://user-images.githubusercontent.com/16258197/184522984-63740ceb-50cc-477c-8f4b-a11aa6aa6137.png" />
</details>

**Comment by dimisa-RUAdList** [comment]
I can reproduce it. These are Google ads, and Adblock Plus considers them acceptable by default. However, this can be easily disabled in the settings: https://forums.lanik.us/viewtopic.php?f=102&t=22511

**Reply by andry81** [reply]
> I can reproduce it. These are Google ads, and Adblock Plus considers them acceptable by default. However, this can be easily disabled in the settings: [forums.lanik.us/viewtopic.php?f=102&t=22511](https://forums.lanik.us/viewtopic.php?f=102&t=22511)

Yes, but it does not show propaganda from VPN.

**Reply by dimisa-RUAdList** [reply]
Google has always made its ads geospecific.

**Reply by andry81** [reply]
You mean that the uBlockOrigin can be enabled to show this type of ad too?

**Reply by dimisa-RUAdList** [reply]
I don't understand your question. No legitimate ad blocker has an ad embedding feature. Ads are shown by visited sites, not blockers.

**Reply by andry81** [reply]
I mean does adblock replaces the ad or it does by the google and so the same behaviour would be for another ad blocker if did happen to left the ad as not removed?

**Reply by dimisa-RUAdList** [reply]
This is what Google does. uBlock Origin blocks this by default. Adblock Plus blocks this if acceptable ads are disabled.

You can check this by disabling the blocker you are using.

**Reply by andry81** [reply]
Is that mean that the web site is responsible to show a specific ad in a region, right? If so, then all the acceptable ads has to be disabled for the sake of security now?

**Reply by dimisa-RUAdList** [reply]
Of course. You can read the full Adblock Plus setup recommendations [here](http://forum.ru-board.com/topic.cgi?forum=5&topic=50761&start=0&limit=1&m=2#1).

**Reply by andry81** [reply]
Thx for explanation!

**Reply by MasterKia** [reply]
@andry81 

_The uBlock project does not support Adblock Plus' "Acceptable Ads Manifesto", because the "Acceptable Ads" marketing campaign is really the business plan of a for-profit entity._

https://github.com/gorhill/uBlock/blob/master/MANIFESTO.md

**Reply by andry81** [reply]
There is an article in russian with deep details about AdBlock Plus marketing: https://habr.com/ru/post/310006/
You can translate the page with the google translation.
